nemo/core/classes/mixins/access_mixins.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC
+from contextlib import contextmanager
+from enum import Enum
+from typing import Dict, Iterator, List, Optional, Union
+
+import torch
+from omegaconf import DictConfig
+
+_ACCESS_CFG = DictConfig({"access_all_intermediate": False, "detach": False, "convert_to_cpu": False})
+
+
+def set_access_cfg(cfg: 'DictConfig'):
+    global _ACCESS_CFG
+    _ACCESS_CFG = cfg
+
+
```

Comment:
This seems good, but needs a way to enable/disable tensor registry - just because the code exists and config exists we should not always register tensors. 

Also handling it via a config flag seems bad cause you would need to update the config and call set_access to update the value. 

What I'd suggest is 
1) remove the config value to enable or disable the registry 
2) setup a global variable with class method to access/update whether the registry should be enabled or not. 
3) in the constructor, call the set config and inside forward of module, call the accessor to see if enabled or not. 
4) config should no longer control whether it's enabled or not though.